src/Blazor.EventGridViewer.ServerApp/Controllers/EventGridController.cs

```diff
@@ -57,10 +58,11 @@ public async Task<IActionResult> Post()
 
                     foreach (EventGridEventModel model in eventGridEventModels)
                     {
+                        EventGridEvent eventGrid = new EventGridEvent(model.Subject, model.EventType, model.DataVersion, model.EventData);
                         // EventGrid validation message
-                        if (model.EventType == EventTypes.EventGridSubscriptionValidationEvent)
+                        if (eventGrid.TryGetSystemEventData(out object systemEvent))
```

Comment:
The systemEvent we get back would still need to be checked to make sure it is the correct type, i.e. SubscriptionValidationEventData. Or we can continue checking the event type, and then do eventGrid.Data.ToObjectFromJson<SubscriptionValidationEventData>().